File: Algorithms/108. Convert Sorted Array to Binary Search Tree/Solution.py
#coding=utf-8
__author__ = 'xuxuan'
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        l=len(nums)
        if l==0:
            return None
        ans=TreeNode(nums[l/2])
        ans.left=self.sortedArrayToBST(nums[0:l/2])
        ans.right=self.sortedArrayToBST(nums[l/2+1:l])
        return ans

File: Algorithms/108. Convert Sorted Array to Binary Search Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return build(nums,0,nums.length-1);
    }
    public TreeNode build(int[] nums,int s,int e){
        if (s>e) {
            return null;
        }
        int m=(s+e)/2;
        int val=nums[m];
        TreeNode node=new TreeNode(val);
        node.left=build(nums,s,m-1);
        node.right=build(nums,m+1,e);
        return node;
    }
}